src/components/app/list-new/AppList.tsx

```diff
@@ -328,29 +328,110 @@ export default function AppList() {
+         *      - If yes then continue.
+         *      - If no then get the Cluster Id from applied filter & push it to updatedAppliedFilters array.
+         */
+        if (filterType === AppListConstants.FilterType.NAMESPACE) {
+            currentlyAppliedFilters.forEach((filter) => {
+                if (!checkedItemIds.some((itemId) => itemId.startsWith(`${filter.split('_')[0]}_`))) {
+                    updatedAppliedFilters.push(filter.split('_')[0]);
+                }
+            });
+        }
+
+        /**
+         * Step 5: Create array out of unique set of item ids & return as string.
+         * - Check & filter out empty items from updatedAppliedFilters array
+         * - Create a new Set out of filtered updatedAppliedFilters array to remove duplicate items.
+         * - Create array out of new Set (as Set doesn't have toString() equivalent) and return as string.
+         */
+        return Array.from(new Set<string>(updatedAppliedFilters.filter((filter) => filter !== ''))).toString();
+    };
+
+    const applyFilter = (type: string, list: FilterOption[], selectedAppTab: string = undefined): void => {
         let qs = queryString.parse(location.search);
         let keys = Object.keys(qs);
         let query = {};
         keys.map((key) => {
             query[key] = qs[key];
-        })
+        });
 
-        let queryParamType = (type == AppListConstants.FilterType.CLUTSER || type == AppListConstants.FilterType.NAMESPACE) ? 'namespace' : type;
-        let checkedItems = list.filter(item => item.isChecked);
-        let ids = checkedItems.map(item => item.key);
-        let str = ids.toString();
-        query[queryParamType] = str;
+        let queryParamType =
+            type == AppListConstants.FilterType.CLUTSER || type == AppListConstants.FilterType.NAMESPACE
+                ? AppListConstants.FilterType.NAMESPACE
+                : type;
+        let checkedItems = list.filter((item) => item.isChecked);
+        let ids = checkedItems.map((item) => item.key);
+
+        query[queryParamType] =
+            queryParamType === AppListConstants.FilterType.NAMESPACE
+                ? getUpdatedFiltersOnNamespaceChange(ids, type, query)
+                : ids.toString();
         query['offset'] = 0;
         query['hOffset'] = 0;
         let queryStr = queryString.stringify(query);
         let _currentTab = selectedAppTab || currentTab;
-        let url = `${_currentTab == AppListConstants.AppTabs.DEVTRON_APPS ? buildDevtronAppListUrl() : buildHelmAppListUrl()}?${queryStr}`;
+        let url = `${
+            _currentTab == AppListConstants.AppTabs.DEVTRON_APPS ? buildDevtronAppListUrl() : buildHelmAppListUrl()
+        }?${queryStr}`;
         history.push(url);
-    }
+    };
 
     const removeFilter = (filter, filterType: string): void => {
-        let val = filter.key;
+        let val = filter.key.toString();
+        let clustId = val.split('_')[0];
```

Comment:
move loacally

Reply:
Can't move it locally as it's used at 2 places so kept it at the top & added a comment as suggested.